src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -690,6 +703,61 @@ private static bool IsNodeOrAnyAncestorLeftSideOfDot(SyntaxNode node, ISyntaxFac
 
             return false;
         }
+
+        internal static ImmutableArray<FindUsageProperty> GetAdditionalProperties(SyntaxNode node, SemanticModel semanticModel, ISyntaxFactsService syntaxFacts)
+        {
+            var additionalProperties = new ArrayBuilder<FindUsageProperty>
+            {
+                GetInfo(syntaxFacts.GetContainingTypeDeclaration(node, node.SpanStart), ContainingTypeInfoPropertyName, semanticModel),
+                GetInfo(syntaxFacts.GetContainingMemberDeclaration(node, node.SpanStart), ContainingMemberInfoPropertyName, semanticModel)
+            };
+            return additionalProperties.ToImmutableAndFree();
+        }
+
+        internal static ImmutableArray<FindUsageProperty> GetAdditionalProperties(ISymbol definition)
+        {
+            var additionalProperties = new ArrayBuilder<FindUsageProperty>();
+
+            var containingType = definition.ContainingType;
+            if (containingType != null)
+            {
+                additionalProperties.Add(GetAdditionalProperty(ContainingTypeInfoPropertyName, containingType));
+            }
+
+            var containingSymbol = definition.ContainingSymbol;
+
+            // Containing member should only include fields, properties, methods, or events.  Since ContainingSymbol can return other types, use the return value of GetMemberType to restrict to members only.
+            if (containingSymbol != null && containingSymbol.GetMemberType() != null)
+            {
+                additionalProperties.Add(GetAdditionalProperty(ContainingMemberInfoPropertyName, containingSymbol));
+            }
+
+            return additionalProperties.ToImmutableAndFree();
+        }
+
+        protected static FindUsageProperty GetInfo(SyntaxNode node, string name, SemanticModel semanticModel)
+        {
+            if (node != null)
+            {
+                var symbol = semanticModel.GetDeclaredSymbol(node);
+                if (symbol != null)
+                {
+                    return GetAdditionalProperty(name, symbol);
+                }
+            }
+
+            return new FindUsageProperty(name, "");
+        }
+
+        private static FindUsageProperty GetAdditionalProperty(string propertyName, ISymbol symbol)
+        {
+            if (symbol == null)
+            {
+                return new FindUsageProperty(propertyName, "");
```

Comment:
Same here - can be converted to a TryGet method instead of returning a dummy property.